File: solutions/101.symmetric-tree/symmetric-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    if(root === null) return true;
    return helper(root.left, root.right);
    
    function helper(p, q){
        if(p === null && q === null){
            return true;
        }else if(p !== null && q !== null){
            if(p.val !== q.val) return false;
            return helper(p.left, q.right) && helper(p.right, q.left);
        }else{
            return false;
        }
    }
    
};

File: solutions/101.symmetric-tree/symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null ) return true;
        return isSameTree(root.left, root.right);
        
    }
    public boolean isSameTree(TreeNode p, TreeNode q){
        if(p == null || q == null){
            if(p == null && q == null){
                return true;
            }else{
                return false;
            }
        }else if(p.val != q.val){
            return false;
        }
        return isSameTree(p.left, q.right) && isSameTree(p.right, q.left);
    }
}